### ISSUE TITLE
[MinGW] invalid build with UNSIGNED_MINGW

### ISSUE THREAD

**Issue by irov** [getsentry/sentry-native#319]
**Description**

Hello!
I try build sentry with MinGW64 (clang) and get error

```
...sentry/src/sentry_sync.c:69:23: error: redefinition of 'DWORD' as different kind of symbol
static UNSIGNED_MINGW DWORD THREAD_FUNCTION_API
                      ^
...msys64\mingw64\x86_64-w64-mingw32\include\minwindef.h:141:29: note: previous definition is here
  typedef unsigned __LONG32 DWORD;
                            ^
...sentry/src/sentry_sync.c:69:28: error: expected ';' after top level declarator
static UNSIGNED_MINGW DWORD THREAD_FUNCTION_API
                           ^
                           ;

```

i found strange code

```
#ifdef __MINGW32__
#    define UNSIGNED_MINGW unsigned
#else
#    define UNSIGNED_MINGW
#endif
```

maybe you set this for 32 bits only? Then need rewrite code to:

`#if defined(__MINGW32__) && !defined(__MINGW64__)`

- OS: Windows 10
- Compiler: msys64/MinGW64

Thanks!

**Comment by Amphaal** [comment]
Why not proposing a pull request ? Seems like a simple and painless change :D

**Comment by irov** [comment]
> Why not proposing a pull request ? Seems like a simple and painless change :D

I am a supporter of the presumption of correctness. I think it was written for some purpose. If what I say is true. Then, of course, I will create a pull request ^_^

**Comment by Amphaal** [comment]
After some digging, I found that at that time, `__MINGW32__` was applied for both 32bits and 64bits builds by mingw-w64 headers. They might have changed that upstream, which is good imo.

So the correct fixed would be something like `#if defined(__MINGW32__) || defined(__MINGW64__)`. It's been quite a while I've not attempted to build Sentry-Native for 64x MinGW 😅.

**Comment by irov** [comment]
> After some digging, I found that at that time, `__MINGW32__` was applied for both 32bits and 64bits builds by mingw-w64 headers. They might have changed that upstream, which is good imo.
> 
> I assume the correct fix would be something like `#if defined(__MINGW32__) || defined(__MINGW64__)`. It's been quite a while I've not attempted to build Sentry-Native for 64x MinGW, so I might be completely off track 😅.

Yes, MINGW32 define for mingw32 and MINGW32 and MINGW64 define for mingw64, like _WIN32 and _WIN64. On Windows 64 bits we have two define _WIN32 and _WIN64. Ok i create pull request :)

**Comment by Swatinem** [comment]
fixed by #321
